hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/volume/OMVolumeSetQuotaRequest.java

```diff
@@ -228,11 +227,24 @@ public boolean checkQuotaBytesValid(OMMetadataManager metadataManager,
     return true;
   }
 
-  public boolean checkQuotaNamespaceValid(long quotaInNamespace) {
-
+  public boolean checkQuotaNamespaceValid(OMMetadataManager metadataManager,
+      long quotaInNamespace, String volumeName) throws IOException {
     if (quotaInNamespace < OzoneConsts.QUOTA_RESET || quotaInNamespace == 0) {
       return false;
     }
+
+    // if volume quota is for reset, no need further check
+    if (quotaInNamespace == OzoneConsts.QUOTA_RESET) {
+      return true;
+    }
+
+    List<OmBucketInfo> bucketList = metadataManager.listBuckets(
+        volumeName, null, null, Integer.MAX_VALUE);
+    if (bucketList.size() > quotaInNamespace) {
```

Comment:
Doesn't volume namespace quota mean the Sum(all bucket namespace quotas) ?
According this code here volume namespace quota  = number of buckets in volume as we are checking `bucketList.size()`

Reply:
@sadanand48 Volume quota is number of buckets it can hold, its not sum of namespace quota for all buckets.

Reply:
Thanks @sumitagrawl , we should document this somewhere , I couldn’t find this info anywhere in code or existing docs, It can get confusing at times as namespace is usually a collection of all objects inside an entity.